framework/src/main/java/org/checkerframework/common/aliasing/AliasingTransfer.java

```diff
@@ -83,15 +85,35 @@ public TransferResult<CFValue, CFStore> visitAssignment(
      */
     @Override
     protected void processPostconditions(
-            MethodInvocationNode n, CFStore store, ExecutableElement methodElement, Tree tree) {
-        super.processPostconditions(n, store, methodElement, tree);
-        if (TreeUtils.isEnumSuper(n.getTree())) {
+            Node n, CFStore store, ExecutableElement methodElement, ExpressionTree tree) {
+        // Process MethodInvocation node only
+        MethodInvocationNode methodInvocationNode = null;
+        ObjectCreationNode objectCreationNode = null;
+        if (n instanceof MethodInvocationNode) {
+            methodInvocationNode = (MethodInvocationNode) n;
+        } else if (n instanceof ObjectCreationNode) {
+            objectCreationNode = (ObjectCreationNode) n;
+        } else {
+            throw new BugInCF(
+                    "Error passing other type of nodes into"
+                            + " processPostconditions of AliasingTransfer");
+        }
+        super.processPostconditions(
+                methodInvocationNode != null ? methodInvocationNode : objectCreationNode,
+                store,
+                methodElement,
+                tree);
+        if (methodInvocationNode != null && TreeUtils.isEnumSuper(methodInvocationNode.getTree())) {
             // Skipping the init() method for enums.
             return;
         }
-        List<Node> args = n.getArguments();
+        List<Node> args =
+                methodInvocationNode != null
+                        ? methodInvocationNode.getArguments()
+                        : objectCreationNode.getArguments();
         List<? extends VariableElement> params = methodElement.getParameters();
-        assert (args.size() == params.size())
+        // Only check MethodInvocationNode
+        assert (methodInvocationNode != null && args.size() == params.size())
```

Comment:
This assert will fail when this method is called for an object creation! Why is this here?

Reply:
Yeah, I need to put this assertion into a if block, thanks!